test/integration/hierarchy-change-auth.int.c

```diff
@@ -0,0 +1,231 @@
+
+	// Init nonce.
+	sessionData.nonce.t.size = 0;
+
+	// init hmac
+	sessionData.hmac.t.size = 0;
+
+	// Init session attributes
+	*( (UINT8 *)((void *)&sessionData.sessionAttributes ) ) = 0;
+
+	sessionsData.cmdAuthsCount = 1;
+	sessionsData.cmdAuths[0] = &sessionData;
+
+
+
+	newAuth.t.size = 0;
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0);
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth for hierarchy 0x%x FAILED! Response Code : 0x%x", tpm_hierarchies[j], rval);
+
+	// Init new auth
+	newAuth.t.size = 20;
+	for( i = 0; i < newAuth.t.size; i++ )
+		newAuth.t.buffer[i] = i;
+
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	// Create hmac session
+	sessionData.hmac = newAuth;
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	// Provide current auth value in SessionData hmac field
+	sessionData.hmac = newAuth;
+	// change auth value to different value
+	newAuth.t.buffer[0] = 3;
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	// Provide current auth value in SessionData hmac field
+	sessionData.hmac = newAuth;
+	// change auth value to different value
+	newAuth.t.buffer[0] = 4;
+	resetAuth = newAuth;
+
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	// Set new auth to zero
+	newAuth.t.size = 0;
+	// Assert that without setting current auth value the command fails
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != (TPM_RC_1 + TPM_RC_S + TPM_RC_BAD_AUTH))
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	sessionsData.cmdAuths[0] = &sessionData;
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != (TPM_RC_1 + TPM_RC_S + TPM_RC_BAD_AUTH))
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	// Set auth to zero again with valid session
+	sessionData.hmac = resetAuth;
+	// change auth value to different value
+	newAuth.t.size = 0;
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, tpm_hierarchies[j], &sessionsData, &newAuth, 0 );
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+	}
+
+	return 0;
+}
+
+/*
+ * Test basic auth value changes for Platform Auth
+ */
+int
+test_0 (TSS2_SYS_CONTEXT *sapi_context)
```

Comment:
Could you rename test_0 and test_1 to something more meaningful, like test_platform_auth() and test_endorsm_auth()